src/libraries/Microsoft.PowerFx.Core/Public/Values/DateTimeValue.cs

```diff
@@ -15,6 +15,46 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public class DateTimeValue : PrimitiveValue<DateTime>
     {
+        /// <summary>
+        /// Direct use of <see cref="Value"/> is prohibited in favor of <see cref="GetConvertedValue(TimeZoneInfo)"/> method.
+        /// </summary>
+        public new DateTime Value => throw new NotSupportedException(string.Format("Use Method {0} instead, with proper timezone information.", nameof(GetConvertedValue)));
+
+        /// <summary>
+        /// Converts To UTC time if value is not utc and <paramref name="timeZoneInfo"/> is utc.
+        /// Converts from UTC time if  value is utc and <paramref name="timeZoneInfo"/> is non utc.
+        /// else returns the  value/>/>.
+        /// NOTE: if <paramref name="timeZoneInfo"/> is null, Local time zone is used.
+        /// </summary>
+        public DateTime GetConvertedValue(TimeZoneInfo timeZoneInfo)
+        {
+            return GetConvertedDateTimeValue(_value, timeZoneInfo);
+        }
+
+        internal static DateTime GetConvertedDateTimeValue(DateTime value, TimeZoneInfo timeZoneInfo)
```

Comment:
>GetConvertedDateTimeValue [](http://example.com/codeflow?start=33&length=25)

Can we adda  unit test for this that covers the various cases? 

I think there's at least 4? 
(value is utc / notUtc) * (timeZoneInfo is utc / notUtc)

Reply:
added.